src/components/common.js

```diff
@@ -314,7 +315,7 @@ export const Select = ({ value, options, id, ...props }) => {
     ...commonSelectProps,
     getOptionLabel: ({ value, label }) => label || value.toString(),
     value: newValue || null, // need null instead of undefined to clear the select
-    options: newOptions
+    options: withGroups ? options : newOptions
```

Comment:
I don't think we want to change this line?

Reply:
The way it's written, we would need to change this line, because the `newOptions` line above flattens out the group structure. However, another approach would be to leave that line alone, and instead change `findValue` so that it understands how to find in groups. That would leave `newOptions` intact, and we could continue to use it here.

Reply:
For the grouped selector `props.options` I'm passing in `options` exactly as the selector is expecting them to be formatted which is why I changed this line. The `newOptions` (when it's `withGroups` ) is a flattened out list of the available options which is needed for getting the correct `value` for the chosen `option`, so then `newOptions` is not correctly formatted to be the `options` property. 

Would it be more straight forward to construct the correctly formatted `options` for a `withGroup` selector inside this function instead?

Reply:
Okay I will look into changing the `findValue` approach rather than changing this line

Reply:
Updated with latest commit